### PR TITLE
:sparkles: Add support for the resource scope marker

### PR DESCRIPTION
**Pull request by droot**
This enables generation of cluster-scoped types.

This carries over the existing https://github.com/kubernetes-sigs/kubebuilder/pull/763 PR.

### CODE REVIEW

File: generated_golden.sh [hunk 1]
```diff
@@ -67,6 +67,7 @@ scaffold_test_project() {
 		$kb create webhook --group crew --version v1 --kind Captain --defaulting --programmatic-validation
 		$kb create api --group crew --version v1 --kind FirstMate --controller=true --resource=true --make=false
 		$kb create webhook --group crew --version v1 --kind FirstMate --conversion
+		$kb create api --group crew --version v1 --kind Admiral --controller=true --resource=true --namespaced=false --make=false
```
Comment (mengqiy) [inline generated_golden.sh]: It seems there is nothing related to `Admiral` got generated.

**Comment by droot** [comment]
@mengqiy addressed review comments. PTAL.

File: testdata/project-v2/config/webhook/manifests.yaml [hunk 2]
```diff
@@ -1,52 +0,0 @@
-
----
```
Comment (mengqiy) [inline testdata/project-v2/config/webhook/manifests.yaml]: Why this file got dropped?
It should be generated by make manifests.

Comment (droot) [reply testdata/project-v2/config/webhook/manifests.yaml]: Strange, I probably didn't run `make generate` properly. Re-ran the whole thing and it seems to be back. Thanks for catching it.

**Comment by mengqiy** [comment]
Linux tests are passing, while macos tests are extremely slow today.
Merging this to unblock the planned release.

**Pull request merged**